plutus-tx/src/PlutusTx/Enum.hs

```diff
@@ -26,8 +26,12 @@ class Enum a where
   toEnum :: Integer -> a
   -- | Convert to an 'Integer'.
   fromEnum :: a -> Integer
-  -- | Construct a list from the given range.
+  -- | Construct a list from the given range (corresponds to [a..b]).
   enumFromTo :: a -> a -> [a]
+  -- | Construct a list from the given range (corresponds to [a,b..c]).  This
+  -- has the same semantics as the Haskell version,so if a==b and c>=b then you
+  -- get an infinite list, which you probably don't want in Plutus Core.
+  enumFromThenTo :: a -> a -> a -> [a]
```

Comment:
I'm not sure if we should add this. It's not that useful and it makes the dictionaries bigger.

Reply:
> I'm not sure if we should add this. It's not that useful and it makes the dictionaries bigger.

Yeah, I wondered about that.  I thought I might as well add it while I was looking at this, but it didn't initially occur to me that it would make the dictionary bigger.